compressed_image_transport/src/compressed_subscriber.cpp

```diff
@@ -92,6 +92,14 @@ void CompressedSubscriber::internalCallback(const sensor_msgs::CompressedImageCo
   // Copy message header
   cv_ptr->header = message->header;
 
+  // Check for an empty input
+  if (message->data.empty())
+  {
+    ROS_WARN_STREAM("Image message was empty.");
```

Comment:
I'm afraid this could spam the logs.

I know there are a bunch of `ROS_ERROR` messages below that could also spam the logs, but this is just a warning.

We should probably throttle it:
```suggestion
    ROS_WARN_STREAM_THROTTLE(30.0, "Image message was empty. This warning message is throttled every 30 seconds.");
```

Reply:
yes i was thinking that as well. thats probably a good idea. this should only happen when running offline so its a little less critical, at least for the current usecase.